src/components/Register.js

```diff
@@ -1,48 +1,103 @@
+      alert(`Request failed. ${error.message}`);
+      throw error;
+    }
+  };
+
+  // Handle register form submission
   const handleRegister = async (e) => {
     e.preventDefault();
     const formData = new FormData(e.target);
     const data = {
-      username: formData.get('username'),
-      password: formData.get('password')
+      username: formData.get("username"),
+      password: formData.get("password"),
     };
 
-    try {
-      const response = await axios.post('https://cicd-2-backend.vercel.app/api/register/', data);
-      console.log('Registration success:', response.data);
-      alert('Registration successful!');
-    } catch (error) {
-      console.error('Registration failed:', error);
-      alert('Registration failed. Please try again.');
-    }
+    await handleSubmit("register/", data, "Registration successful!");
+  };
+
+  // Handle login form submission
+  const handleLogin = async (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    const data = {
+      email: formData.get("email"),
+      password: formData.get("password"),
+    };
+
+    await handleSubmit("login/", data, "Login successful!");
+  };
+
+  // Toggle between login and register tabs
+  const toggleTab = () => {
+    setActiveTab(activeTab === "login" ? "register" : "login");
+    document.getElementById("chk").checked = activeTab === "login";
   };
 
   return (
     <>
-      <title>Login</title>
+      <title>Account Access</title>
       <link rel="stylesheet" type="text/css" href="style.css" />
       <link
         href="https://fonts.googleapis.com/css2?family=Jost:wght@500&display=swap"
         rel="stylesheet"
       />
       <div className="main">
-        <input type="checkbox" id="chk" aria-hidden="true" />
+        <input
+          type="checkbox"
+          id="chk"
+          aria-hidden="true"
+          checked={activeTab === "register"}
+          onChange={toggleTab}
+        />
         <div className="signup">
           <form onSubmit={handleRegister}>
             <label htmlFor="chk" aria-hidden="true">
               Register
             </label>
-            <input type="text" name="username" placeholder="Username" required />
-            <input type="password" name="password" placeholder="Password" required />
+            <input
+              type="text"
+              name="username"
+              placeholder="Username"
+              required
+            />
+            <input
+              type="password"
+              name="password"
+              placeholder="Password"
+              required
+            />
             <button type="submit">Register</button>
           </form>
         </div>
+        <div className="login">
+          <form onSubmit={handleLogin}>
+            <label htmlFor="chk" aria-hidden="true">
+              Login
+            </label>
+            <input type="email" name="email" placeholder="Email" />
```

Comment:
this field should be required.
```suggestion
            <input type="email" name="email" placeholder="Email" required />
```

Reply:
fixed ty!